tests/fourier/test_spectrum.py

```diff
@@ -0,0 +1,216 @@
+            return qml.expval(qml.PauliZ(wires=0))
+
+        res = spectrum(circuit, encoding_gates=["x"])(0.1)
+        expected_degree = n_qubits * n_layers
+        assert np.allclose(res["x"], range(-expected_degree, expected_degree + 1))
+
+    def test_spectrum_changes_with_qnode_args(self):
+        """Test that the spectrum changes per call if a qnode argument changes the
+        circuit architecture."""
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit(last_gate):
+            qml.RX(0.1, wires=0, id="x")
+            qml.RX(0.2, wires=1, id="x")
+            if last_gate:
+                qml.RX(0.3, wires=2, id="x")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        res_true = spectrum(circuit)(True)
+        assert np.allclose(res_true["x"], range(-3, 4))
+
+        res_false = spectrum(circuit)(False)
+        assert np.allclose(res_false["x"], range(-2, 3))
+
+    def test_input_gates_not_of_correct_form(self):
+        """Test that an error is thrown if gates marked as encoding gates
+        are not single-parameter gates."""
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.1, wires=0, id="x")
+            qml.Rot(0.2, 0.3, 0.4, wires=1, id="x")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        with pytest.raises(ValueError, match="can only consider one-parameter gates"):
+            spectrum(circuit)()
+
+
+def circuit(x, w):
+    """Test circuit"""
+    for l in range(2):
+        for i in range(3):
+            qml.RX(x[i], wires=0, id="x" + str(i))
+            qml.RY(w[l][i], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+    qml.RZ(x[0], wires=0, id="x0")
+    return qml.expval(qml.PauliZ(wires=0))
+
+
+expected_result = {
+    "x0": [-3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0],
+    "x1": [-2.0, -1.0, 0.0, 1.0, 2.0],
+    "x2": [-2.0, -1.0, 0.0, 1.0, 2.0],
+}
+
+
+class TestInterfaces:
+    """Test that inputs are correctly identified and spectra computed in
+    all interfaces."""
+
+    def test_integration_autograd(self):
+        """Test that the spectra of a circuit is calculated correctly
+        in the autograd interface."""
+
+        x = pnp.array([1, 2, 3], requires_grad=False)
+        w = pnp.array([[-1, -2, -3], [-4, -5, -6]], requires_grad=True)
+
+        dev = qml.device("default.qubit", wires=3)
+        qnode = qml.QNode(circuit, dev, interface="autograd")
+
+        res = spectrum(qnode)(x, w)
+        for (k1, v1), (k2, v2) in zip(res.items(), expected_result.items()):
+            assert k1 == k2
+            assert v1 == v2
+
+    def test_integration_torch(self):
+        """Test that the spectra of a circuit is calculated correctly
+        in the torch interface."""
+
+        torch = pytest.importorskip("torch")
+        x = torch.tensor([1.0, 2.0, 3.0, 4.0, 5.0], requires_grad=True)
```

Comment:
Why are the `x` in the non-autograd interfaces different than the one in autograd? The circuit itself looks like it needs only a length 3 array.

Reply:
Oh that was a leftover...